server/channels/store/sqlstore/store.go

```diff
@@ -455,32 +471,93 @@ func (ss *SqlStore) GetSearchReplicaX() *sqlxDBWrapper {
 		return ss.GetReplicaX()
 	}
 
-	rrNum := atomic.AddInt64(&ss.srCounter, 1) % int64(len(ss.searchReplicaXs))
-	return ss.searchReplicaXs[rrNum]
+	cnt := 0
+	for {
+		rrNum := atomic.AddInt64(&ss.srCounter, 1) % int64(len(ss.searchReplicaXs))
+		if ss.searchReplicaXs[rrNum].Load().Online() {
+			return ss.searchReplicaXs[rrNum].Load()
+		}
+		cnt++
+		// If all replicas are down, then go with replica.
```

Comment:
I didn't get this comment, can you elaborate?

Reply:
Sorry, this should be "If all search replicas are down, then go with replica"